optaplanner-operator/optaplanner-operator-integration-test-parent/optaplanner-operator-integration-test/src/test/java/org/opaplanner/operator/integration/OptaPlannerSolverReconcilerIT.java

```diff
@@ -0,0 +1,140 @@
+
+    @Test
+    public void optaPlannerSolverReconcilerIT() {
+
+        createCrAmqBroker(solverName);
+
+        OptaPlannerSolver solver = new Yaml(new Constructor(OptaPlannerSolver.class))
+                .load(OptaPlannerSolverReconcilerIT.class.getClassLoader()
+                        .getResourceAsStream(SCHOOL_TIMETABLING_SOLVER_YML));
+        solver.getSpec().getAmqBroker().setHost("ex-aao-amqp-0-svc." + solverName + ".svc.cluster.local");
+        solver.getSpec().getAmqBroker().setManagementHost("ex-aao-hdls-svc." + solverName + ".svc.cluster.local");
+        solver.getMetadata().setName(solverName);
+        solver.getMetadata().setNamespace(solverName);
+        kubernetesClient.resources(OptaPlannerSolver.class).inNamespace(solverName).create(solver);
+
+        final String expectedMessageAddressIn = solver.getInputMessageAddressName();
+        final String expectedMessageAddressOut = solver.getOutputMessageAddressName();
+
+        await().ignoreException(NullPointerException.class).atMost(1, MINUTES).untilAsserted(() -> {
+            OptaPlannerSolver updatedSolver = kubernetesClient
+                    .resources(OptaPlannerSolver.class)
+                    .inNamespace(solver.getMetadata().getNamespace())
+                    .withName(solver.getMetadata().getName())
+                    .get();
+
+            assertThat(updatedSolver.getStatus()).isNotNull();
+            assertThat(updatedSolver.getStatus().getConditions()).isNotEmpty();
+            Condition condition = updatedSolver.getStatus().getConditions().get(0);
+            assertStatusCondition(condition, updatedSolver, OptaPlannerSolverStatus.ConditionStatus.TRUE);
+
+            assertThat(updatedSolver.getStatus().getInputMessageAddress())
+                    .isEqualTo(expectedMessageAddressIn);
+            assertThat(updatedSolver.getStatus().getOutputMessageAddress())
+                    .isEqualTo(expectedMessageAddressOut);
+        });
+
+        ConfigMap configMap = kubernetesClient
+                .resources(ConfigMap.class)
+                .inNamespace(solver.getMetadata().getNamespace())
+                .withName(solver.getConfigMapName())
+                .get();
+        Map<String, String> configMapData = configMap.getData();
+        assertThat(configMapData.get(ConfigMapDependentResource.SOLVER_MESSAGE_INPUT_KEY))
+                .isEqualTo(expectedMessageAddressIn);
+        assertThat(configMapData.get(ConfigMapDependentResource.SOLVER_MESSAGE_OUTPUT_KEY))
+                .isEqualTo(expectedMessageAddressOut);
+
+        List<Deployment> deployments = kubernetesClient
+                .resources(Deployment.class)
+                .inNamespace(solver.getMetadata().getNamespace())
+                .list()
+                .getItems();
+        assertThat(deployments).hasSize(1);
+        assertThat(deployments.get(0).getMetadata().getName()).isEqualTo(solverName);
+    }
+
+    private void createCrAmqBroker(String namespace) {
```

Comment:
For the "full scenarios": if the test creates a broker, it should also wait for it to start up before sending any message.